### PR TITLE
Adopt @skyux/list-builder-common

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "@skyux/i18n": "^3.3.0",
     "@skyux/indicators": "^3.0.0-rc.2",
     "@skyux/layout": "^3.0.0-rc.1",
+    "@skyux/list-builder-state": "3.0.0-alpha.0",
```
Comment (blackbaud-conorwright) [inline package.json]: needs a carot "^"

Comment (Blackbaud-AlexKingman) [reply package.json]: good catch. fixed.

File: src/app/public/modules/list/state/items/item.model.ts [hunk 2]
```diff
@@ -1,15 +1,3 @@
...
-      throw new Error('All list item models require an ID');
-    }
-
-    this.id = id;
-    if (data !== undefined) {
-      this.data = data;
-    }
-  }
-}
+export {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/list/state/items/item.model.ts]: These exports (and elsewhere) can be outright deleted since we're still in RC.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list/state/items/item.model.ts]: deleted

File: src/app/public/modules/list/helpers.ts [hunk 3]
```diff
@@ -2,34 +2,9 @@ import {
...
-    return undefined;
-  }
-
-  return result;
-}
+export {
+  getData
+} from '@skyux/list-builder-common/helpers';
 
 export function compare(value1: any, value2: any) {
```
[the remark this answers is not in the document]
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list/helpers.ts]: done.

File: src/app/public/modules/list/state/index.ts [hunk 4]
```diff
@@ -1,10 +1,14 @@
+export {
+  ListItemModel,
+  ListSortFieldSelectorModel
+} from '@skyux/list-builder-common';
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/list/state/index.ts]: Can we remove this export so that there's no confusing where the common stuff comes from?

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list/state/index.ts]: Good catch - I should have caught that. Fixed.

**Pull request merged**